### ISSUE TITLE
Getting error "Method does not override any method from its superclass" from generated output when using maps

### ISSUE THREAD

**Issue by DeepAnchor** [alexeyxo/protobuf-swift#109]
Hey,

Thanks for creating this project!

I have a simple message definition with map types like so:

```
syntax = "proto3";

message ParticipantWrapper {
  map<string, Participant> participant = 1;
}

message Participant {
  uint32 id = 1;
  string name = 2;
}
```

When I compile it (using the Protobuf3.0-Swift2.0 branch), the generated output shows many errors with "Method does not override any method from its superclass". I don't think I've used any reserved keywords so it does not appear to be related to #34. Idea idea what the problem could be?

**Comment by alexeyxo** [comment]
What version runtime library you have use?

**Comment by DeepAnchor** [comment]
I believe it's version 1.6 from the `ProtoBuf3.0-Swift2.0` branch

**Comment by albeebe** [comment]
This may be unrelated but I pulled the latest code and got compilation errors about overriding a superclass and the solution was to upgrade to the latest protobuf-swift protoc plugin.

**Comment by DeepAnchor** [comment]
hey, thanks for the info @albeebe, yeah I'm pretty sure i'm doing something wrong on my end since the latest travis build seems to be passing. Did you mean you updated the swift runtime or the compiler? 
If you're talking about the runtime, which version worked for you?

I tried running it with a clean build in xcode with the latest runtime (2.4.1) and it showed these errors.

**Comment by albeebe** [comment]
Yesterday I upgraded the runtime to the current code in the repo and I got compilation errors about overriding the superclass. It wasn't until later on in the day I upgraded to the latest version of the compiler that the errors went away. Right now I'm using the most up to date runtime and compiler and things are working. 

Up until today I was using a copy of the runtime and compiler that I installed back in October.

**Comment by albeebe** [comment]
@deepanchor I should note that I'm using the code from the master branch

**Comment by DeepAnchor** [comment]
hmm, interesting. Just to clarify, are you using `syntax="proto3"`?

**Comment by albeebe** [comment]
no proto2 is what I'm using

**Comment by alexeyxo** [comment]
I think you use compiler from "Proto3.0-Swift2.0" branch, but runtime library from master.
You should use compiler and library from "Proto3.0-Swift2.0" branch. Your example is working in my test project.

**Comment by DeepAnchor** [comment]
Ok, the problem is solved now. I think it had something to do with the cocoapods cache referring to the master branch instead of the `Proto3.0-Swift2.0` branch, even though I had clearly installed the runtime from the `Proto3.0-Swift2.0` branch. Very odd indeed, since I had tried clearing the cache and reloading the workspace several times.

What I ended up doing was just deleting everything related to cocoapods, clearing *all* cached pods as well as removing the `Pods/` folder and just starting with a fresh `pod install`.

Thanks for all your help!
